app/templates/suppliers/registered_address.html

```diff
@@ -0,0 +1,147 @@
+        "label": "Supplier details"
+      },
+    ]
+  %}
+    {% include "toolkit/breadcrumb.html" %}
+  {% endwith %}
+{% endblock %}
+
+
+{% block main_content %}
+  {% if error %}
+  <div class="validation-masthead" aria-labelledby="validation-masthead-heading">
+    <h3 class="validation-masthead-heading" id="validation-masthead-heading">
+      {{ error }}
+    </h3>
+  </div>
+  {% endif %}
+
+  {% if registered_address_form.errors or registered_country_form.errors %}
+      <div class="validation-masthead" aria-labelledby="validation-masthead-heading">
+          <h3 class="validation-masthead-heading" id="validation-masthead-heading">
+              There was a problem with the details you gave for:
+          </h3>
+          {% if registered_address_form.errors %}
+          <ul>
+          {% for field_name, field_errors in registered_address_form.errors|dictsort %}
+            {% for error in field_errors %}
+              <li><a href="#{{ field_name }}" class="validation-masthead-link">{{ registered_address_form[field_name].label.text }}</a>
+            {% endfor %}
+          {% endfor %}
+          </ul>
+          {% endif %}
+
+          {% if registered_country_form.errors %}
+          <ul>
+          {% for field_name, field_errors in registered_country_form.errors|dictsort %}
+            {% for error in field_errors %}
+            <li><a href="javascript:document.getElementById('location-autocomplete').focus()" class="validation-masthead-link">{{ registered_country_form[field_name].label.text }}</a></li>
+            {% endfor %}
+          {% endfor %}
+          </ul>
+          {% endif %}
+      </div>
+  {% endif %}
+
+  <div class="grid-row">
+    <div class="column-one-whole">
+      {% with
+        heading = "What is your registered office address?",
+        smaller = true
+      %}
+        {% include 'toolkit/page-heading.html' %}
+      {% endwith %}
+    </div>
+  </div>
+
+  <form action="{{ url_for('.update_registered_address') }}" method="post" enctype="multipart/form-data">
+
+  <input type="hidden" name="csrf_token" value="{{ csrf_token() }}" />
+
+  <div class="grid-row">
+    <div class="column-two-thirds">
+
+      {{ forms.input('id', registered_address_form.id.data, type='hidden') }}
+      {{ forms.question_input('address1', 'Building and street', value=registered_address_form.address1.data, errors=registered_address_form.address1.errors) }}
+      {{ forms.question_input('city', 'Town or city', value=registered_address_form.city.data, errors=registered_address_form.city.errors) }}
+      {{ forms.question_input('postcode', 'Postcode', value=registered_address_form.postcode.data, errors=registered_address_form.postcode.errors) }}
+
+      {% if registered_country_form.registrationCountry.errors %}
+      <div class="validation-wrapper">
+        <style>
+          #location-autocomplete {border: 5px solid #B10E1E;}
+          .autocomplete__hint {border: 5px solid;}
+        </style>
```

Comment:
The reason we've done this is due to the JavaScript the country picker is running. When the JS loads it adds a new input field and gives it the id of `location-autocomplete`. 
Because the input is dynamically generated we can't conditionally assign it a class in the template, so we need to target and update that elements style here.
It's the same for the `autocomplete__hint` class which is defined in a stylesheet brough in by the node package for country picker. We can't edit the existing class or apply a new class to the element we need to (again dynamically generated) so we need to override it here.

Reply:
I guess I'm being dense. If those static styles just lived in application.scss, wouldn't they be applied when the elements get dynamically created by the location picker?

Reply:
No, it is a bit confusing.

We do have some static styles in the application.scss to already override these styles (adds some padding to make the country picker match our other input fields). The issue is that when that field has validation errors, those styles need to be overridden in a different way (increasing the border size and applying the validation colour). Normally in the template we'd have an if statement which applied a new class (with these new styles) to the element when there were validation errors. But the element is not defined in our template, it's generated dynamically on page load by the javascript. So we can't apply a new class to it, so have to update the styles that are already applied.

Reply:
Righto, totally follow this now. Thanks for explaining. 👍